### ISSUE TITLE
Improve melee weapon generation

### ISSUE THREAD

**Issue by angelusmetal** [angelusmetal/dungeon#170]
- Tweak trait scores
- Tweak range & speed; type should depend on that (e.g. daggers have lower range, but higher speed; slower weapons have more reach; some weapons have higher spread, which may land blows that don't do much but others that do a lot).
- Think about whether material affects anything - not damage (otherwise all high level weapons will look the same)
- Weapon effects (not just name; also affect color)

**Comment by angelusmetal** [comment]
I did a nice approximation of different weapon types and materials (each affecting weight, speed, how much energy is drained with each blow). There is no "better" weapon type; all of them have pros and cons and scale with weapon score/level. Gold cost does as well.

Also I did a first implementation of fire effect (it can be stacked) by making traits both expirable now, and stackable (they were a fixed set before) and in the meantime, found & fixed a nasty concurrent modification bug.
